### ISSUE TITLE
The Gatsby plugin I built builds properly locally, but not when used via an NPM package

### ISSUE THREAD

**Issue by njosefbeck** [gatsbyjs/gatsby#3722]
## Description

Hello! I created a Gatsby plugin back in October that inserted the Stripe JS onto each page of a Gatsby site, similar to how your Google Analytics plugin worked. Code repo can be found [here](https://github.com/njosefbeck/gatsby-plugin-stripe-elements). I haven't touched this code for a couple of months, but was notified in December of issues occurring when users of my plugin try to run `gatsby build` and `gatsby develop`.

I recently decided to tackle this issue, and indeed thought I had fixed it. Turns out, I wasn't importing React in gatsby-ssr.js as I should have been.) You can see this commit [here](https://github.com/njosefbeck/gatsby-plugin-stripe-elements/commit/72c6dbff5950ba7eb823c3c90f623c1e7aef24b4). After adding React, I verified that everything built fine locally, committed my changes, and re-published my NPM package.

Unfortunately, users are still having issues with the build process when using my plugin. I also experience issues when I'm using the NPM version of my plugin instead of when I have it directly in my `plugins` directory. I'm at a loss as to why this could be happening, as the code is the same, regardless of if I have it directly in my `plugins` directory if I've installed it via NPM.

Here's a screenshot of the error I'm getting:

![screen shot 2018-01-26 at 7 11 35 pm](https://user-images.githubusercontent.com/9508002/35466819-15c0a412-02cd-11e8-8c67-9b0c2d55302a.png)

My only thought as to why it could be failing is because maybe it doesn't recognize the <script> tags as a proper React element, but if that were the case, I don't understand why it wouldn't fail when the plugin is in the `plugins` directory as well.

I'm not sure how much help you'll be able to provide, but any ideas or things to test would be much appreciated!

Thanks!

### Environment

Gatsby version: 1.9.158
Node.js version: 8.9.4
Operating System: Mac OS High Sierra

### File contents (if changed):
`gatsby-ssr.js`: 
```javascript
import React from 'react';

exports.onRenderBody = ({ setPostBodyComponents }, pluginOptions) => {
  return setPostBodyComponents([
    <script
      key={`gatsby-plugin-stripe-elements`}
      src="https://js.stripe.com/v3/"
    >
    </script>
  ]);
}
```

### Actual result

See above.

### Expected behavior

Gatsby should build fine, and should inject the <script> tag reference in gatsby-ssr into every page on the Gatsby site.

### Steps to reproduce

1\. Follow the steps here to add to your Gatsby site: https://github.com/njosefbeck/gatsby-plugin-stripe-elements

2\. Run `gatsby build` or `gatsby develop`.

**Comment by njosefbeck** [comment]
Additionally, I tried changing the React element syntax like the below, but that also didn't work.

```javascript
<script 
  key={`gatsby-plugin-stripe-elements`}
  dangerouslySetInnerHTML={{ __html: 'src="https://js.stripe.com/v3/"' }}
/>
```

**Comment by shannonbux** [comment]
Just saw this issue, @njosefbeck! Thanks for the detailed write-up :) There is a new [plugin authoring page on gatsbyjs.org](https://www.gatsbyjs.org/docs/plugin-authoring/). Wondering if it addresses your issue--@jlengstorf any ideas on why he might be getting this issues?

**Comment by jlengstorf** [comment]
@njosefbeck have you done a diff between the plugins version and the node_modules version to make sure they're identical?

If they're identical, then I'm not sure what's up. It shouldn't be possible for the same code to work differently, so this will be a weird one to track down. Let us know if that didn't come back clean.

**Comment by njosefbeck** [comment]
@shannonbux @jlengstorf I was able to fix it, had to do with needing to compile the gatsby-ssr code using babel. Will close this - thanks for the replies!

**Comment by njosefbeck** [comment]
Nah, this was more on my end actually! I think your docs are just fine :) but thanks for checking!

**Comment by raae** [comment]
I had the same issue. 

I have developed a plugin as a local plugin, it works fine. Then I pushed it to GitHub and tried using it as a dependency through `npm install <git-host>:<git-user>/<repo-name>` and it fails because of the script tag.

I feel that "how to get your plugin ready for the ecosystem" is not very well documented.

**Comment by shannonbux** [comment]
@raae what do you think would help people get plugins ready for the ecosystem?

**Comment by raae** [comment]
How you go from a working local plugin to a plugin others can test to one that is available in the plugin directory at Gatsby. 

In this case, I was wondering if there were any best practices regarding babel transpiling. But after talking to Micheal yesterday I realized I could skip that babel and just use React.createElement in gatsby-ssr.js.

After this was fixed my plugin can be tested by others using `npm install <git-host>:<git-user>/<repo-name>`. I found this to be a relief as I feel the plugin is not ready to become "offical" yet.

I guess I'll know more when I have finished the completed the whole process.
